### PR TITLE
feat(auth): user gql query

### LINKED ISSUE #4049: [auth][graphql] `getUsers` resolver
```graphql
query users {
  {
    username
    email
    role
  }
}
```

Note this resolver should not return passwords. Only admins should be able to get user information (should use permission from #4050). Probably don't need pagination right away.

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #4049 

```graphql
{
  users {
    edges {
      node {
        username
        email
      }
    }
  }
}
```

### CODE REVIEW

File: src/phoenix/server/api/queries.py [hunk 1]
```diff
@@ -71,6 +72,35 @@
...
+        after: Optional[CursorString] = UNSET,
+        before: Optional[CursorString] = UNSET,
+    ) -> Connection[User]:
+        args = ConnectionArgs(
+            first=first,
+            after=after if isinstance(after, CursorString) else None,
+            last=last,
+            before=before if isinstance(before, CursorString) else None,
+        )
+        stmt = select(models.User).order_by(models.User.email)
```
Comment (axiomofjoy) [inline src/phoenix/server/api/queries.py:90]: Probably need to omit the system user.

**Review by axiomofjoy — APPROVED** [review]
We probably also want to display the user's role in addition.

File: src/phoenix/server/api/queries.py (hunk 1 above)
Comment (mikeldking) [reply src/phoenix/server/api/queries.py:90]: Yeah let's figure out these nuances as we go. For now I just want to get the plumbing setup.

**Comment by mikeldking** [comment]
> We probably also want to display the user's role in addition.

Yup, just doing small PRs for now to de-risk

**Pull request merged**